code/game/objects/effects/anomalies.dm

```diff
@@ -265,13 +265,25 @@
 		T.atmos_spawn_air("o2=5;plasma=5;TEMP=1000")
 
 /obj/effect/anomaly/pyro/detonate()
+	INVOKE_ASYNC(src, .proc/makeslime)
+
+/obj/effect/anomaly/pyro/proc/makeslime()
 	var/turf/open/T = get_turf(src)
 	if(istype(T))
 		T.atmos_spawn_air("o2=500;plasma=500;TEMP=1000") //Make it hot and burny for the new slime
-
 	var/new_colour = pick("red", "orange")
 	var/mob/living/simple_animal/slime/S = new(T, new_colour)
-	S.rabid = 1
+	S.rabid = TRUE
+	S.amount_grown = SLIME_EVOLUTION_THRESHOLD
+	S.Evolve()
+
+	var/list/mob/dead/observer/candidates = pollCandidatesForMob("Do you want to play as [S.name]?", null, null, 0, 300, S, null)
+	var/mob/dead/observer/chosen = null
+
+	if(candidates.len)
+		chosen = pick(candidates)
+		message_admins("[key_name_admin(chosen)] was spawned as [S.name]")
+		S.key = chosen.key
```

Comment:
pollCandidatesForMob just polls deadchat and returns a list of potential candidates. offer_control does that AND actually assigns the mob and logs it and informs the admin